### PR TITLE
Feature/4606

### LINKED ISSUE #4606: 打印插件不能正常打印footer部分，或者mergecell的也不能正常打印。
打印插件不能正常打印footer部分，或者mergecell的也不能正常打印。

### PR DESCRIPTION
**Pull request by UtechtDustin**
**Bug fix?**
no

**New Feature?**
yes

**Resolve an issue?**
Fix #4606

**Example(s)?**
Before: https://live.bootstrap-table.com/code/UtechtDustin/3469
After: https://live.bootstrap-table.com/code/UtechtDustin/3473

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -2553,6 +2553,16 @@ class BootstrapTable {
     }
 
     $td.attr('rowspan', rowspan).attr('colspan', colspan).show()
+    if (this.mergedCells === undefined) {
```
Comment (djhvscf) [inline src/bootstrap-table.js:2556]: Can we use if (!this.mergedCells) {...

File: src/extensions/print/bootstrap-table-print.js [hunk 2]
```diff
@@ -126,20 +126,71 @@ $.BootstrapTable = class extends $.BootstrapTable {
 
       html.push('</thead><tbody>')
 
+      const dontRender = []
+      if (this.mergedCells !== undefined) {
```
Comment (djhvscf) [inline src/extensions/print/bootstrap-table-print.js:130]: Can we use if (this.mergedCells)

**Comment by djhvscf** [comment]
The print result is not corresponding with the table itself. There are lot of misalignments

**Comment by UtechtDustin** [comment]
> The print result is not corresponding with the table itself. There are lot of misalignments

Do you have an example for that ?
For me it looks the same, without any misalignments (Im also not sure how this can happens, because while printing we just build a normal table without special styling etc.).

**Comment by UtechtDustin** [comment]
@djhvscf reminder :-)

**Comment by djhvscf** [comment]
image

**Comment by UtechtDustin** [comment]
For me its working fine, which setup do you use @djhvscf ?

**Comment by djhvscf** [comment]
Nothing special. Using Chrome latest version and windows

**Comment by UtechtDustin** [comment]
Okay, its a issue of chrome.
I will check that!

**Comment by UtechtDustin** [comment]
@djhvscf Fixed for chrome :-)

**Comment by wenzhixin** [comment]
@djhvscf improved the code, please review.

**Pull request merged**